src/map/battle.cpp

```diff
@@ -4667,19 +4801,25 @@ static int battle_calc_attack_skill_ratio(struct Damage* wd, struct block_list *
 			RE_LVL_DMOD(120);
 			break;
 		case LG_CANNONSPEAR:
-			skillratio += -100 + skill_lv * (50 + sstatus->str);
+			skillratio += -100 + (120 + sstatus->str) * skill_lv;
+			if (sc && sc->data[SC_SPEAR_SCAR])
+				skillratio += 400;
 			RE_LVL_DMOD(100);
 			if (sc && sc->data[SC_SPEAR_SCAR])// Whats the official increase? [Rytech]
 				skillratio += skillratio * 50 / 100;
```

Comment:
Is this unintended duplicate skillratio addition? If so, which `SC_SPEAR_SCAR` bonus are we keeping?